pkg/kubecost/allocationfilter.go

```diff
@@ -100,6 +101,10 @@ type AllocationFilter interface {
 	Flattened() AllocationFilter
 
 	String() string
+
+	// Equals returns true if the two AllocationFilters are logically
+	// equivalent.
+	Equals(AllocationFilter) bool
```

Comment:
Does this really work for logical equivalence? ie: `!(a && b) == (!a || !b)` 

We may not have negation support yet, but just wanted to make sure.

Reply:
I'm pretty sure this works for logical equivalence right now because we only support negation at the leaf level: `a != b` is supported but `not(a == b)` is not supported. This gets much thornier if/when we introduce higher-level negation. Does that address the question? I'm racking my brain for other gotchas of logical equivalence.

Reply:
Yeah, no worries  - just being picky.